drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1919,15 +1958,14 @@ static void eth0_iface_init(struct net_if *iface)
 		LOG_ERR("ETH PHY Initialization Error");
 		return;
 	}
-	/* PHY auto-negotiate link parameters */
-	result = phy_sam_gmac_auto_negotiate(&cfg->phy, &link_status);
-	if (result < 0) {
-		LOG_ERR("ETH PHY auto-negotiate sequence failed");
-		return;
-	}
 
-	/* Set up link parameters */
-	link_configure(cfg->regs, link_status);
+	/* Initialise monitor */
+	k_delayed_work_init(&dev_data->monitor_work, monitor_work_handler);
+	k_delayed_work_submit(&dev_data->monitor_work,
```

Comment:
Given that this API has the word "work" in it, I'm assuming its high priority ? If so, it would be a bad idea to use it as the PHY auto-neg process can take up a long time and the work being high priority will hang up the system albeit temporarily. 

If all I've said makes sense, I will recommend to make it a thread with configurable priority so the user can set it based on their needs.

Reply:
I would suggest that we **do not** create a specific thread just for this purpose but utilize the system workqueue like is done here.

Reply:
Auto-neg always takes in the order of seconds and that's why I recommended this approach. Is this approach not preferred because its somewhat memory intensive, i.e. dedicated stack for the separate thread ? 
If we go with the work-based approach and the work has higher priority over other system tasks (which it does), we will see the system hang for the duration of auto-negotiation (1-N secs). 
In most of the cases, we should have a link-up/down interrupt line between the MAC / PHY chips and so this will not be required.

Reply:
> Given that this API has the word "work" in it, I'm assuming its high priority ?

No, the system workqueue is often the place where you dump all the tasks that aren't time-critical and can run later.

Reply:
> I would suggest that we **do not** create a specific thread just for this purpose but utilize the system workqueue like is done here

Yes, that was the point. Other Zephyr Ethernet drivers that already have an "rx_thread" poll the link status there. For GMAC, there isn't one and I wanted to avoid creating a thread just for that as much as possible.

Reply:
> > Given that this API has the word "work" in it, I'm assuming its high priority ?
> 
> No, the system workqueue is often the place where you dump all the tasks that aren't time-critical and can run later.

Great.. For where I come, work is the highest priority execution context from the scheduler point-of-view. That's why I said "if my understanding is correct".. Overall this is better than creating a task..